multishard_mutation_query.cc

```diff
@@ -578,28 +578,17 @@ future<> read_context::lookup_readers(db::timeout_clock::time_point timeout) noe
             auto& semaphore = this->semaphore();
             auto shard = this_shard_id();
 
-            auto querier_opt = db.get_querier_cache().lookup_shard_mutation_querier(cmd->query_uuid, *schema, *ranges, cmd->slice, gts.get(), timeout);
+            auto querier_opt = db.get_querier_cache().lookup_shard_mutation_querier(cmd->query_uuid, *schema, *ranges, cmd->slice, semaphore, gts.get(), timeout);
 
             if (!querier_opt) {
                 _readers[shard] = reader_meta(reader_state::inexistent);
                 co_return;
             }
 
-            auto& q = *querier_opt;
-
-            if (&q.permit().semaphore() != &semaphore) {
-                co_await q.close();
-                on_internal_error(mmq_log, format("looked-up reader belongs to different semaphore than the one appropriate for this query class: "
-                        "looked-up reader belongs to {} (0x{:x}) the query class appropriate is {} (0x{:x})",
-                        q.permit().semaphore().name(),
-                        reinterpret_cast<uintptr_t>(&q.permit().semaphore()),
-                        semaphore.name(),
-                        reinterpret_cast<uintptr_t>(&semaphore)));
-            }
-
             // At this point the readers is passed to the semaphore and we are
             // safe w.r.t. exceptions. The code between this point and obtaining
             // the querier must take care of closing it if an error happens.
```

Comment:
This comment can be deleted as well. No one is throwing exception between here and the lookup anymore.